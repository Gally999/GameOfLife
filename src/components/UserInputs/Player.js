import React from 'react';
import PropTypes from 'prop-types';

const Player = ({ steps, onStepsUpdate, onEvolveClick }) => {
  const updateValue = (event) => {
    onStepsUpdate(parseInt(event.target.value, 10));
  };

  return (
    <div>
      <div>
        <label htmlFor="steps">Please choose the number of evolution steps:
          <div />
          <input
            type="number"
            value={steps}
            id="steps"
            name="steps"
            min={1}
            onChange={updateValue}
          />
        </label>
      </div>
      <button
        type="button"
        onClick={onEvolveClick}
      >
        Evolve
      </button>
    </div>
  );
};

Player.propTypes = {
  steps: PropTypes.number.isRequired,
  onEvolveClick: PropTypes.func.isRequired,
  onStepsUpdate: PropTypes.func.isRequired,
};

export default Player;
