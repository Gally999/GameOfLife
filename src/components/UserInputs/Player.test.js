import React from 'react';
import Player from './Player';

describe('Player', () => {
  it('renders the options component correctly', () => {
    const subject = (
      <Player
        steps={4}
        onStepsUpdate={() => {}}
        onEvolveClick={() => {}}
      />
    );
    expect(subject).toMatchSnapshot();
  });
});
