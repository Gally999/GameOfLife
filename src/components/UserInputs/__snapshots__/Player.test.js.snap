// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders the options component correctly 1`] = `
<Player
  onEvolveClick={[Function]}
  onStepsUpdate={[Function]}
  steps={4}
/>
`;
